### PR TITLE
[Feature] Implement LoadingIndicator Component

### PR DESCRIPTION
**Pull request by narin**
**NOTE:** Needs #348 to be merged and a new assets package published before merging

## Description of Change

Adds a LoadingIndicator component based on [Figma design](https://www.figma.com/file/UlUUAKULuL2peMxSvfY5wx/%5BNEW%5D-Loading-indicator---%23249?type=design&node-id=264-686&mode=dev)
- Uses newly added loading_indicator icon added to the assets page
- Regenerated the icon list using the handy new script
- Animates rotation of the component using react native built in animations
- Accepts text and optional children. `text` is required if no children are provided

#### Testing Packages

components
- 0.17.1-alpha.0
- 0.17.1-alpha.1
- 0.17.1-alpha.2

#### iOS Storybook

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/e5cc32f9-48a3-4a75-9dfc-cf55fd32bee4

#### iOS In-App

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/62d2ada7-fa23-424f-8b2b-023663066f6b

#### Android Storybook

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/b52e3495-5ff1-48af-b3b6-fa99313def63

#### Android In-App

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/dc80e38a-70e2-42a1-89e3-9b8e82578e06

#### Web

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/4a5ce9e3-eb6a-471b-bfa3-06df1871272d

## Testing

- [x] Tested on iOS 
- [x] Tested on Android 
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish

If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy) and the PR is approved and ready to merge:
- [ ] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

### CODE REVIEW

File: packages/components/src/components/SegmentedControl/SegmentedControl.stories.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { SegmentedControl, SegmentedControlProps } from './SegmentedControl'
 import { generateDocs } from '../../utils/storybook'
 
 const meta: Meta<SegmentedControlProps> = {
-  title: 'Segmented control',
+  title: 'SegmentedControl',
```
Comment (TimRoe) [inline packages/components/src/components/SegmentedControl/SegmentedControl.stories.tsx:9]: Why was this changed?

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import { View } from 'react-native'
+import React from 'react'
+
+import { Link } from '../Link/Link'
+import { LoadingIndicator, LoadingIndicatorProps } from './LoadingIndicator'
+import { Spacer } from '../Spacer/Spacer'
+import { generateDocs } from '../../utils/storybook'
+
+const meta: Meta<LoadingIndicatorProps> = {
+  title: 'LoadingIndicator',
```
Comment (TimRoe) [inline packages/components/src/components/LoadingIndicator/LoadingIndicator.stories.tsx:11]: I'd think this should be the external facing name per Figma:  `Loading indicator`.

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          justifyContent: 'center',
+          alignItems: 'center',
+        }}>
+        <Story />
+      </View>
+    ),
+  ],
+  parameters: {
+    docs: generateDocs({
+      name: 'LoadingIndicator',
```
Comment (TimRoe) [inline packages/components/src/components/LoadingIndicator/LoadingIndicator.stories.tsx:27]: Similar comment to above.

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  }
+
+  return (
+    <View style={containerStyle}>
+      <Animated.View style={indicatorStyle}>
+        <Icon {...iconProps} />
+      </Animated.View>
+      {text && (
+        <>
+          <Spacer />
```
Comment (TimRoe) [inline packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx:96]: Nitpicky, but may be worth pulling the Spacer out since one will always be present and just conditionalizing the second Spacer to flatten things out a la:
```
</Animated.View>
<Spacer />
{text && <Text>{text}</Text>}
{text && children ? <Spacer /> : null}
{children ? children : null}
```

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    transform: [{ rotate }],
+    justifyContent: 'center',
+    alignItems: 'center',
+  }
+
+  const iconProps: IconProps = {
+    name: 'LoadingIndicator',
+    width: 50,
+    height: 50,
+    fill: isDarkMode ? Colors.uswdsBlueVivid30 : Colors.primary,
```
Comment (TimRoe) [inline packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx:78]: Looking at Figma, I think we now have semantic tokens to use for these--this is set to `vads-color-action-foreground-default`.

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  }
+
+  const textStyle: TextStyle = {
+    fontFamily: 'SourceSansPro-Regular',
+    fontSize: 20,
+    lineHeight: 30,
+    textAlign: 'center',
+    color: isDarkMode ? Colors.grayLightest : Colors.base,
```
Comment (TimRoe) [inline packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx:86]: Similar comment to above, but `vads-color-foreground-default`.

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx (hunk 5 above)
Comment (narin) [reply packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx:78]: Will need to merge [@jessicawoodin's PR first](https://github.com/department-of-veterans-affairs/va-mobile-library/pull/349) but I agree

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  })
+
+  const containerStyle: ViewStyle = {
+    width: '100%',
+    alignItems: 'center',
+  }
+
+  const indicatorStyle: ViewStyle = {
+    width: 50,
+    height: 50,
```
Comment (TimRoe) [inline packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx:68]: Were these needed or are they redundant with the iconProps sizing below?

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    transform: [{ rotate }],
+    justifyContent: 'center',
+    alignItems: 'center',
```
Comment (TimRoe) [inline packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx:71]: Similar question: were these needed or are they redundant since it just hugs the Icon within it?

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx (hunk 5 above)
Comment (TimRoe) [reply packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx:78]: Not certain, but in theory your branch updating to the same version hers does shouldn't cause conflict if it moved first. Planning to look at her PR next though.

**Review by TimRoe — CHANGES_REQUESTED** [review]
Looks good but for a couple minor outstanding questions.

Marking request changes just to denote end of review/needing to update assets package before merging.

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.stories.tsx (hunk 2 above)
Comment (narin) [reply packages/components/src/components/LoadingIndicator/LoadingIndicator.stories.tsx]: Reverted. Wasn't aware we were using the figma naming in our stories

File: packages/components/src/components/SegmentedControl/SegmentedControl.stories.tsx (hunk 1 above)
Comment (narin) [reply packages/components/src/components/SegmentedControl/SegmentedControl.stories.tsx]: Reverted

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx (hunk 4 above)
Comment (narin) [reply packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx]: Simplified the spacing here

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx (hunk 5 above)
Comment (narin) [reply packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx]: Updated tokens but linting/tests will fail for other components while her PR is unmerged

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx (hunk 6 above)
Comment (narin) [reply packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx]: Updated to use semantic tokens

File: packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx (hunk 7 above)
Comment (narin) [reply packages/components/src/components/LoadingIndicator/LoadingIndicator.tsx]: Good catch. These stylings aren't needed.
